File: googleanalytics.php
<?php

/**
    Google Analytics
 *  @author:  Willem Labu
 *            Flint n Tinder
 *
 *  Google's Mobile Analytics was deprecated.
 *  Rewriting it as per the guidelines here: http://goo.gl/ii3Bvl  
 */


// Your UA code
$GA_ACCOUNT = "UA-XX-X";

// Using this in a MXit app?
$MXIT = true;

/**
 *  A unique identifier for the current user
 *  http://goo.gl/EeHnoo
 */
$UID = '';


/**
 *  A switch
 *
  true
 *
 *  BLOCKING
 *   Best to include this at the _bottom_ of your page.
 *
 *  Uses `file_get_contents`
 *  Make sure your server has `allow_url_fopen` turned on.
 *
  false
 *
 *  NON-BLOCKING
 *   Best to include this at the _top_ of your page.
 *
 *  Uses `exec` & `cURL`
 *  Make sure your server has `php_curl` installed and on.
 */
$BLOCKING_REQUEST = false;




/**************************************
 *   NO NEED TO EDIT ANYTHING BELOW   *
 **************************************/

// The current page URL
$PAGE = $_SERVER["REQUEST_URI"];

// GA's URL to POST to
$GA_URL = "http://www.google-analytics.com/collect";


// MXit specific overwrites
$UID = isset($_SERVER["HTTP_X_MXIT_USERID_R"]) ? md5($_SERVER["HTTP_X_MXIT_USERID_R"]) : $UID;
$UA  = isset($_SERVER["HTTP_X_DEVICE_USER_AGENT"]) ? $_SERVER["HTTP_X_DEVICE_USER_AGENT"] : $_SERVER['HTTP_USER_AGENT'];
$MXIT_PIXELS = isset($_SERVER["HTTP_UA_PIXELS"]) ? $_SERVER["HTTP_UA_PIXELS"] : '';


// Here we go!
if ($BLOCKING_REQUEST) {

  $data = array(
    'payload_data' => '',
    'v'     => '1',
    't'     => 'pageview',
    'dp'    => $PAGE,
    'tid'   => $GA_ACCOUNT,

    'cid'   => $UID,
    'uid'   => $UID,
    'ua'    => $UA,
    'sr'    => $MXIT_PIXELS
  );

  $options = array(
      'http' => array(
          'header'  => "Content-type: application/x-www-form-urlencoded\r\n",
          'method'  => 'POST',
          'content' => http_build_query($data),
      ),
  );

  $context  = stream_context_create($options);

  // Hit!
  echo file_get_contents($GA_URL, false, $context);


} else {

  $url = "$GA_URL?payload_data&v=1&t=pageview&dp=" .
          urlencode($PAGE) . "&tid=$GA_ACCOUNT&cid=$UID&uid=$UID&ua=" .
          urlencode($UA)   . "&sr=$MXIT_PIXELS";

  $cmd = "curl '" . $url . "' > /dev/null 2>&1 &";

  // Hit!
  exec($cmd, $output, $exit);

}

?>

File: GoogleAnalytics.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net;
using System.Text;
using System.Web;
using System.Web.UI;

namespace yournamespace
{


/**
    Google Analytics
 *  @author:  Emmanuel Matsiyana
 *            Flint n Tinder
 *
 *  Google's Mobile Analytics was deprecated.
 *  Rewriting it as per the guidelines here: http://goo.gl/ii3Bvl  
 */
    public class GoogleAnalytics
    {

        // Global function for tracking pageviews and events

        public void GoogleAnalyticsTracking(String Hostname, String Path, String pageTitle, String trackType, String eventCategory, String Action, String Label)
        {


            var GaAccount = "UA-XXXX-Y";
            var request = (HttpWebRequest)WebRequest.Create("http://www.google-analytics.com/collect");
            var hostname = Hostname; // Request.Url.GetLeftPart(UriPartial.Authority);
            var path = Path;        // HttpUtility.UrlEncode(Request.Url.PathAndQuery);
            var pagetitle = pageTitle;
            var userCookie = GetRandomNumber();

            var postData = "v=1";                   //Version
            postData += "&tid=" + GaAccount;    //Tracking ID / Property ID
            postData += "&cid=" + userCookie;   //Client/User ID
            postData += "&t=" + trackType;      //Track type ? Pageview or Event


            if (trackType == "pageview")
            {

                postData += "&dh=" + hostname;       // Document hostname.
                postData += "&dp=" + path;           // Page.
                postData += "&dt=" + pagetitle;      // Title.

            }
            else if (trackType == "event")
            {

                postData += "&ec=" + eventCategory;      // Event Category. Required.
                postData += "&ea=" + Action;         // Event Action. Required.
                postData += "&el=" + Label;         // Event label.
                postData += "&ev=1";                // Event value.

            }

            var data = Encoding.ASCII.GetBytes(postData);

            request.Method = "POST";
            request.ContentType = "application/x-www-form-urlencoded";
            request.ContentLength = data.Length;

            using (var stream = request.GetRequestStream())
            {
                stream.Write(data, 0, data.Length);
            }

            var response = (HttpWebResponse)request.GetResponse();

            var responseString = new StreamReader(response.GetResponseStream()).ReadToEnd();


        }

        private static String GetRandomNumber()
        {
            Random RandomClass = new Random();
            return RandomClass.Next(0x7fffffff).ToString();
        }

    }
}